contracts/IQLF.sol

```diff
@@ -12,14 +12,25 @@ import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
 
 abstract contract IQLF is IERC165 {
     /**
-     * @dev Returns if the given address is qualified, implemented on demand.
+     * @dev Check if the given address is qualified, implemented on demand.
+     *
+     * Requirements:
+     *
+     * - `account` account to be checked
+     * - `proof` data to prove if a user is qualified.
+     *           For instance, it can be a MerkelProof to prove if a user is in a whitelist
+     *
+     * Return:
+     *
+     * - `bool` whether the account is qualified for ITO
+     * - `string` if not qualified, it contains the error message(reason)
      */
-    function ifQualified (address account) virtual external view returns (bool);
+    function ifQualified (address account, bytes32[] memory proof) virtual external view returns (bool, string memory);
```

Comment:
I'd prefer to change the `proof` to `data` to generalize its usage.

Reply:
updated